### PR TITLE
Better configuration flow, limiting number of elements and collections, snackbar messages

### PR DESCRIPTION
**Pull request by arkadR**
TODO: Tests

### CODE REVIEW

File: Main/ClientApp/src/Components/Dialogs/ChangeMaxCollectionsDialog.tsx [hunk 1]
```diff
@@ -21,33 +22,29 @@ export default function ChangeMaxCollectionsDialog(
...
       actions={
         <Button
           disabled={
-            maxCollections === props.maxCollections ||
-            ((maxCollections as unknown) as number) < 0
+            maxCollections === props.maxCollections || maxCollections <= 0
           }
           onClick={() => {
-            changeValue("MaxCollections", maxCollections);
+            props.onValueChange(maxCollections);
```
Comment (karol7531) [inline Main/ClientApp/src/Components/Dialogs/ChangeMaxCollectionsDialog.tsx:47]: i think that this component responsibility should be changing value of property "MaxCollections", so component should know what to do, but now the action is injected and this action may not change the property value.  if component would have been more abstract this approach would be good, but now it isn't

File: Main/Controllers/ConfigurationController.cs [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
+﻿using LinkCollectionApp.Data;
+using LinkCollectionApp.Infrastructure.DTO;
+using LinkCollectionApp.Infrastructure.Interfaces;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Authorization;
 
 namespace LinkCollectionApp.Controllers
 {
-  [Authorize(Roles = "Administrator")]
+  // [Authorize(Roles = "Administrator")]
```
Comment (karol7531) [inline Main/Controllers/ConfigurationController.cs:9]: why // ?

Comment (arkadR) [reply Main/Controllers/ConfigurationController.cs:9]: woops

File: Main/ClientApp/src/Components/Dialogs/ChangeMaxCollectionsDialog.tsx (hunk 1 above)
Comment (arkadR) [reply Main/ClientApp/src/Components/Dialogs/ChangeMaxCollectionsDialog.tsx:47]: The goal is to make it more abstract

**Pull request merged**